### DISCUSSION TITLE
Great tool for developers!

### DISCUSSION THREAD

**Discussion by mickeyl** [collin80/SavvyCAN#318] [Show and tell]
Let me shell out a big fat **THANKS** to @collin80 and all contributors. Right now my task is developing an UDS-based ECU reprogramming software for iOS and SavvyCAN is making my live so much easier, since I can thoroughly inspect the CAN logs taken from another software for analyzing them.

I'm using SavvyCAN on Linux (for capturing the logs) and on macOS (for analyzing) and on both platforms, it was a breeze to compile and run.

**Comment by FrankM54** [comment]
Hey mickeyl,
Is there much setup required to compile it on a Mac? 
Right now I am using an Ubuntu VM, but native MacOS would be nice.

I do have a little xcode experience.

**Reply by mickeyl** [reply]
Super-simple, no `Xcode` involved. Here's what I did:

1. Install [Homebrew](https://brew.sh): `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`
2. Install Qt5: `brew install qt@5`
3. Checkout this git repository and `cd` into
4. Create the makefiles via QMake: `/usr/local/Cellar/qt@5/5.15.2/bin/qmake`
5. Compile: `make`
6. Run the finished binary.

**Reply by FrankM54** [reply]
Wow, that worked perfectly the first time!
Thank you for putting together that list.

Quick question, how do I make a stand alone app package, so it can run on an other Mac?

**Reply by mickeyl** [reply]
This, unfortunately, is not trivial, as there is no full static linking on macOS. If you can‘t install Qt via homebrew, then  the target machine needs to have the Qt .dylibs installed to somewhere your DYLD_LIBRARY_PATH points to. Apart from that, you can take the compiled SavvyCAN app and copy it over.

**Reply by collin80** [reply]
There is actually a way. Back when I was actually building MacOS binaries (sorry... kind of lapsed on that...) I used macdeployqt which comes with QT for MacOS. You run macdeployqt and point it at your app folder for the SavvyCAN binary you built. Then it copies all the proper libraries into the app folder. From there you can move that app folder to another Mac and run SavvyCAN.

https://doc.qt.io/qt-5/macos-deployment.html

**Reply by FrankM54** [reply]
Thanks collin80!
I will give it a go :)

**Reply by vincentbarat** [reply]
Using macdeployqt works, but unfortunately, there is an issue with QT SerialBus devices plugins.
I'm currently trying to fix this...

**Reply by vincentbarat** [reply]
I got it working!

Here is the procedure I used to build a self contained version of SavvyCAN for macOS supporting Peak CAN-USB plugin:

0 - Pre-requesite: QT 5.15.5 must be installed (this procedure was tested using HowmeBrew: $ brew install qt@5)

1 - Download and install the MacCAN driver 0.11.2
$ wget https://github.com/mac-can/PCBUSB-Library/releases/download/v0.11.2/macOS_Library_for_PCANUSB_v0.11.2.tar.gz
$ tar xvzf macOS_Library_for_PCANUSB_v0.11.2.tar.gz
$ cd macOS_Library_for_PCANUSB_v0.11.2
$ make install

2 - Download and compile QTSerialBus 5.15.5
$ git clone https://github.com/qt/qtserialbus.git
$ cd qtserialbus
$ git checkout git checkout v5.15.5-lts-lgpl
$ qmake
$ make

3 - Download and compile the latest version of SavvyCAN
$ git clone https://github.com/collin80/SavvyCAN.git
$ cd SavvyCAN
$ qmake
$ make

4 - Build SavvyCAN bundled with all plugins and libraries includes
$ macdeployqt SavvyCAN.app
$ cp -r ../qtserialbus/plugins/canbus SavvyCAN.app/Contents/PlugIns/
$ macdeployqt SavvyCAN.app

Note that macdeployqt is used twice: first for the app itself, then for the plugins

**Reply by ShemSinghCyruss** [reply]
Hi I am able to compile and do stuffs but unable to configure port connected with PEAKCAN USB
Can you please help me out?

**Comment by shanlanCoding** [comment]
Hello, do you have any recommended UDS tools? I want to test my new energy vehicle exhaustively, but there are too many messages, with 3000 frames per second.

**Comment by gsegov** [comment]
Hello! I don't know if my question is for this topic but I will give a try.
Is it possible Savycan to read binary files and translate it to human read format (example: If I would like to understand in wich ID is the DPF sensors)?
